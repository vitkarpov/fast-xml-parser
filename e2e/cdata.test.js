const fs = require('fs');
const path = require('path');
const {parse, stringify} = require('../dst/index');

describe('cdata', () => {
  const input = fs.readFileSync(path.join(__dirname, 'inputs/cdata.html'), 'utf-8');

  it('parse', () => {
    expect(parse(input)).toMatchSnapshot();
  });

  it('stringify', () => {
    expect(stringify(parse(input))).toMatchSnapshot();
  });
});
