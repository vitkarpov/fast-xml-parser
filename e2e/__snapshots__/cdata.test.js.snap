// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cdata parse 1`] = `
Node {
  "attrs": Object {},
  "children": Array [
    Node {
      "attrs": Object {},
      "children": Array [],
      "name": "!DOCTYPE html",
      "type": "DIRECTIVE",
    },
    Node {
      "attrs": Object {},
      "children": Array [
        Node {
          "attrs": Object {},
          "children": Array [],
          "name": "Within this Character Data block I can use double dashes as much as I want (along with <, &, ', and \\")",
          "type": "CDATA_TAG",
        },
      ],
      "name": "html",
      "type": "OPENING_TAG",
    },
  ],
  "name": "root",
  "type": "FRAGMENT",
}
`;

exports[`cdata stringify 1`] = `"<!DOCTYPE html><html><![CDATA[Within this Character Data block I can use double dashes as much as I want (along with <, &, ', and \\")]]></html>"`;
